src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+
+                Dim variable = variables.First()
+                If parentExpression Is Nothing Then
+                    expression = variable.Expression
+
+                    If variable.AsClause IsNot Nothing Then
+                        Dim identifier = SyntaxFactory.IdentifierName(NameOf(Enumerable.Cast))
+                        Dim generic = SyntaxFactory.GenericName(
+                            identifier.Identifier,
+                            SyntaxFactory.TypeArgumentList(SyntaxFactory.SingletonSeparatedList(variable.AsClause.Type)))
+                        expression = SyntaxFactory.InvocationExpression(
+                            SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            expression,
+                            SyntaxFactory.Token(SyntaxKind.DotToken), generic),
+                            SyntaxFactory.ArgumentList())
+                    End If
+
+                    Return expression
+                End If
+
+                Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression)
+                Dim firstLambda = CreateLambdaExpression(anonymousFunctions.First())
+                Dim secondArgumentAnonymousFunction = anonymousFunctions.Last()
+                Dim returnedValue = DirectCast(secondArgumentAnonymousFunction.Body.Operations.First(), IReturnOperation).ReturnedValue
+                Dim secondLambda As LambdaExpressionSyntax
+                If returnedValue.Kind = OperationKind.AnonymousObjectCreation Then
+                    Dim objectCreation = DirectCast(returnedValue, IAnonymousObjectCreationOperation)
+                    Dim fieldInitializers = objectCreation.Initializers.
+                        Select(Function(initializer) SyntaxFactory.InferredFieldInitializer(
+                        SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name))))
+                    Dim anonymousObjectCreation = SyntaxFactory.AnonymousObjectCreationExpression(
+                        SyntaxFactory.ObjectMemberInitializer(SyntaxFactory.SeparatedList(Of FieldInitializerSyntax)(fieldInitializers)))
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction, anonymousObjectCreation)
+                Else
+                    secondLambda = CreateLambdaExpression(secondArgumentAnonymousFunction)
+                End If
+
+                Return CreateInvocationExpression(expression, NameOf(Enumerable.SelectMany), {firstLambda, secondLambda})
+            End Function
+
+            Private Function CreateLambdaExpression(anonymousFunction As IAnonymousFunctionOperation, Optional body As VisualBasicSyntaxNode = Nothing) As LambdaExpressionSyntax
+                If body Is Nothing Then
+                    body = MakeIdentifierNameReplacements(anonymousFunction.Body.Operations.First().Syntax)
```

Comment:
>anonymousFunction.Body.Operations.First().Syntax [](start = 58, length = 48)

Is this a compiler synthesized return statement? I would not take dependency on is syntax, we are not making any guarantees for nodes like this. If there is a syntax from source, use that, otherwise translate IOperation tree without references back to syntax.